### PR TITLE
docs: update for Athens 3

### PR DESCRIPTION
**Pull request by fadeev**
* updated tutorials to use the temporary branch of `example-contracts` with Athens 3 support
* moved the pros and cons of CCM/Omnichain to the "Start building with ZetaChain section"
* created a top-level "Architecture" section that will host blockchain and protocol contracts docs
* removed examples that duplicate the existing tutorials (like "Your first zEVM contract")
* added Multioutput tutorial

### CODE REVIEW

File: docs/developers/cross-chain-messaging/examples/cross-chain-counter.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+📜 Contract address: 0x2A5831b69EbBD6941e68A17bAD34184B7c4a7628
+
+🔗 Setting interactors for a contract on mumbai_testnet
+✅ Interactor address for 97 (bsc_testnet) is set to 0x2a5831b69ebbd6941e68a17bad34184b7c4a7628
+
+🔗 Setting interactors for a contract on bsc_testnet
+✅ Interactor address for 80001 (mumbai_testnet) is set to 0xb10bd8b51b433079e960ec91beadcdc4434e3e69
+```
+
+## Increment the counter value
```
Comment (btroo) [inline docs/developers/cross-chain-messaging/examples/cross-chain-counter.md:86]: maybe this title should be "Test the deployment" or something similar?

seems this `Increment the counter value` is a subheading below

File: docs/developers/omnichain/bitcoin.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+:::
+
+To deposit BTC into zEVM ZRC-20 contract (and optionally call a smart contract),
+the Bitcoin transaction must conform to this specifications:
+
+1. The Bitcoin transaction must have at least 2 outputs;
+2. The first output must be addressed to the TSS Bitcoin address;
+3. The second output must be a memo output, i.e. `OP_RETURN PUSH_x [DATA]`. This
+   output must be less than 80 bytes;
```
Comment (btroo) [inline docs/developers/omnichain/bitcoin.md:20]: `;` -> `.`

File: docs/developers/omnichain/bitcoin.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+4. The memo `[DATA]` is an array of bytes that encodes the recipient address of
```
Comment (btroo) [inline docs/developers/omnichain/bitcoin.md:21]: perhaps we could have real examples of what `[DATA]` should look like for 4-6 here. i think it'd help people understand better what they need to use/encode for various use cases!

File: docs/developers/omnichain/bitcoin.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+Where `recipient` is the TSS Bitcoin address, `amount` is the amount of tBTC to
+transfer, and `memo` is the recipient address on zEVM.
+
+## Example 2: Deposit BTC and call a smart contract in zEVM
+
+:::tip
+
+In order to test with Bitcoin, you will need to use a wallet that allows setting
```
Comment (btroo) [inline docs/developers/omnichain/bitcoin.md:62]: may also want to clarify that the wallet must support a binary memo input. not essential but may help people troubleshoot.

File: docs/developers/omnichain/overview.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+in-depth example.
+
+An omnichain contract is deployed on ZetaChain and can be called from any
+connected chain.
+
+To call on omnichain contract the only thing a user has to do is send a
+transaction to a connected chain to ZetaChain's TSS address. The transaction
+`amount` becomes available to the sender on ZetaChain and the `data` byte array
+(containing an the omnichain contract `address` and `message`) is used to call
+the omnichain contract by `address` and pass arguments from the `message`.
```
Comment (btroo) [inline docs/developers/omnichain/overview.md:48]: we could  add a sentence like `Check out ZRC-20 to learn how to implement this.` that links to the relevant ZRC-20 doc (and/or others)

File: docs/developers/omnichain/tutorials/single-input-multiple-output.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+sidebar_position: 5
+---
+
+# Single Input Multiple Output
+
+## Overview
+
+If you already read the previous tutorials you already know how to use zEVM. A
+very common use case on zEVM is a smart contract with a single input from one
+chain, do some magic, and then execute the output to another or multiple chains.
```
Comment (btroo) [inline docs/developers/omnichain/tutorials/single-input-multiple-output.md:11]: maybe `do some logic` -> `perform some logic`

And we could add a sentence like `This capability may be useful for applications like multichain asset managers or DeFi applications that need to distribute or manage assets on many chains from one place.`

File: docs/developers/omnichain/zrc-20.md [hunk 7]
```diff
@@ -63,18 +63,17 @@ extremely simple interface to also function in an omnichain way.
...
-module) address.
+([Testnet](/reference/testnet), [Mainnet](/reference/mainnet)) on a connected
+chain, `deposit` is called by `zetacore` and made available to the address that
+deposited. If there is data on the TX `message`, the system contract
+`DepositAndCall` is called, forwarding that data in a call to `onCrossChainCall`
+on the target zEVM contract. The `deposit` and `DepositAndCall` functions are
+only callable by the CCTX module (`zetacore` module) address.
 
 This is a snippet of what the system contract looks like, where `DepositAndCall`
 may be called by `zetacore` after receiving a dep osit into a TSS address
```
Comment (btroo) [inline docs/developers/omnichain/zrc-20.md:74]: `dep osit` -> `deposit`

File: docs/developers/overview.mdx [hunk 8]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+referred to as a Threshold Signature Scheme (TSS) address.
+
+- If the transfer only includes **an asset**, the asset becomes available to the
+  omnichain contract.
+- If the transfer includes **an asset and a message** containing the omnichain
+  contract address and data, the asset becomes available to the omnichain
+  contract, and the contract is called with the message data as an argument.
+
+Assets transferred to a TSS address on ZetaChain are represented as ZRC-20
+tokens. ZRC-20 is an extension of ERC-20 that supports additional functionality
```
Comment (btroo) [inline docs/developers/overview.mdx:50]: we could link to zrc-20 here

File: docs/developers/overview.mdx [hunk 9]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+
+- Transfer of native gas asset (ETH, BNB, MATIC, etc.) between connected chains.
+- Transfer of ERC-20 tokens between connected chains.
+
+### Advantages of omnichain contracts
+
+Developers can extend common applications like Uniswap V2/V3, Curve, Aave, and
+Compound by making them compatible with ZRC-20. These applications have been
+audited and battle-tested on Ethereum and can easily be deployed and built on
+ZetaChain.The changes required to add ZRC-20 support are minimal, and users may
```
Comment (btroo) [inline docs/developers/overview.mdx:65]: `ZetaChain.The` -> `ZetaChain. The`

File: docs/developers/overview.mdx [hunk 10]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+interact with these applications in single-step transactions just as they would
+on Ethereum.
+
+ZRC-20 can easily support Bitcoin/Algorand/Cardano/XRP which do not have
```
Comment (btroo) [inline docs/developers/overview.mdx:69]: `Bitcoin/Algorand/Cardano/XRP` -> `Bitcoin/Dogecoin`

File: docs/developers/overview.mdx [hunk 11]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+through a unified pool in a single transaction.
+
+With ZRC-20/zEVM, exception/revert handling is much simpler as foreign
+interaction is confined to either a standard ERC-20/contract interaction
+(success) or no contract interaction (failure) immediately.
+
+## Cross-chain message passing
+
+With cross-chain message (CCM) passing you deploy **a CCM-enabled contract on
+each connected chain**.
```
Comment (btroo) [inline docs/developers/overview.mdx:91]: `that can pass arbitrary message data and value between each other.`?

File: docs/developers/overview.mdx [hunk 12]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+messages. For example, you could trade Ethereum ETH directly for Polygon USDC
+through a unified pool in a single transaction.
+
+With ZRC-20/zEVM, exception/revert handling is much simpler as foreign
+interaction is confined to either a standard ERC-20/contract interaction
+(success) or no contract interaction (failure) immediately.
+
+## Cross-chain message passing
+
+With cross-chain message (CCM) passing you deploy **a CCM-enabled contract on
```
Comment (btroo) [inline docs/developers/overview.mdx:90]: we can link to the CCM page here?

File: docs/developers/overview.mdx [hunk 13]
```diff
@@ -23,8 +23,92 @@ modify states across different chains in a single step.
...
+CCM supports any type of data transfer and it's up to the contracts to decide
+how to handle the data.
+
+CCM is a reliable general purpose solution for augmenting existing applications
+with cross-chain functionality.
+
+With messages, you can leverage existing liquidity like Uniswap pools on
+existing chains, and trade via burning/minting ZETA through ZetaChain. This
+approach may be more complex (more transactions involved, more gas), but doesn't
+rely on liquidity within ZetaChain's ecosystem.
```
Comment (btroo) [inline docs/developers/overview.mdx:114]: add-on sentence:

`By leveraging the ZETA burn/mint functionality built into ZetaChain, one can build value-transfer applications that don't bridge/wrap assets, minimizing risk for users.`

**Review by btroo — APPROVED** [review]
looking great! just left a few minor comments for copy improvements.

File: docs/developers/omnichain/bitcoin.md (hunk 2 above)
Comment (fadeev) [reply docs/developers/omnichain/bitcoin.md]: 80ac0c2

File: docs/developers/omnichain/bitcoin.md (hunk 3 above)
Comment (fadeev) [reply docs/developers/omnichain/bitcoin.md:21]: Good idea! I think for users to also be able to test it out it should be part of `send-btc` task. Created an issue to add it once `send-btc` is a bit more user-friendly in terms of memo-handling.

https://github.com/zeta-chain/docs/issues/61

File: docs/developers/omnichain/bitcoin.md (hunk 4 above)
Comment (fadeev) [reply docs/developers/omnichain/bitcoin.md:62]: 6cd83db

File: docs/developers/omnichain/overview.md (hunk 5 above)
Comment (fadeev) [reply docs/developers/omnichain/overview.md]: 78ad898

File: docs/developers/omnichain/zrc-20.md (hunk 7 above)
Comment (fadeev) [reply docs/developers/omnichain/zrc-20.md]: 32ccbd6

File: docs/developers/overview.mdx (hunk 8 above)
Comment (fadeev) [reply docs/developers/overview.mdx]: 7b621ca

File: docs/developers/overview.mdx (hunk 9 above)
Comment (fadeev) [reply docs/developers/overview.mdx]: def78ac

File: docs/developers/overview.mdx (hunk 10 above)
Comment (fadeev) [reply docs/developers/overview.mdx]: b56ba8d

File: docs/developers/overview.mdx (hunk 11 above)
Comment (fadeev) [reply docs/developers/overview.mdx]: 20a1924

File: docs/developers/overview.mdx (hunk 12 above)
Comment (fadeev) [reply docs/developers/overview.mdx]: 2086eaf

File: docs/developers/overview.mdx (hunk 13 above)
Comment (fadeev) [reply docs/developers/overview.mdx:118]: 857346b

**Comment by fadeev** [comment]
Thanks, @btroo 🙏 I think I've resolved all the comments.

**Review by btroo — APPROVED** [review]

**Pull request merged**